acvm/src/compiler/transformers/fallback.rs

```diff
@@ -32,6 +32,13 @@ impl FallbackTransformer {
                     acir_supported_opcodes.push(opcode);
                     continue;
                 }
+                Opcode::MemoryInit { .. } | Opcode::MemoryOp { .. } => {
+                    if !is_supported(&opcode) {
+                        return Err(CompileError::UnsupportedOpcode(opcode.to_enum()));
```

Comment:
I think this is a bit misleading, since we have Arithmetic as a variant and this should never return the arithmetic variant because Arithmetic is always supported

Reply:
I have renamed the enum and make sure it can only be called for these two opcodes.